src/util/upload-plan-parser.ts

```diff
@@ -21,32 +22,47 @@ class UploadPlanParser {
         }
     }
 
-    static convertToDownloadFilesJob(job: Job, downloadDirBasePath: string): DownloadFilesJob {
+    static convertToDownloadFilesJob(job: Job, downloadDirBasePath: string): DownloadS3FilesJob {
         const filesToDownload: File[] = job.conversion ? job.conversion.inputs : job.files;
 
-        const downloadJob: DownloadFilesJob = {
+        return {
             basePath: downloadDirBasePath,
             container: job.manifest_id,
             files: UploadPlanParser.parseDownloadFiles(filesToDownload)
-        };
+        }
+    }
 
-        return downloadJob
+    static convertToDownloadBundleFilesJob(job: Job, basePath: string): DownloadBundleFilesJob {
+        const filesToDownload: File[] = job.conversion ? job.conversion.inputs : job.files;
+        if (job.dcp_bundle_uuid) {
```

Comment:
This check happens twice, I think we should remove the check here. This function should assume it has well-formed, non-null parameters. I would pass in the bundle_uuid as a non-nullable String param, then the check is unnecessary and the absence of job.dcp_bundle_uuid is instead caught by the type system.